src/pipecat/processors/frame_processor.py

```diff
@@ -96,6 +96,10 @@ def __init__(
         self.__input_event = asyncio.Event()
         self.__input_frame_task: Optional[asyncio.Task] = None
 
+        # Handle to cancel the tasks gracefully
+        self.__input_frame_task_cancelling = False
+        self.__push_frame_task_cancelling = False
```

Comment:
In theory, invoking `task.cancel()` schedules a `CancelledError` to be thrown at the next `await` inside the coroutine, allowing it to exit gracefully.

But we've observed some rare cases where this does not happen as expected, and the task may hang during cancellation. Like if it's stuck in synchronous code, not hitting an await, or suppressing the `CancelledError`.

So I am adding this as a safeguard mechanism to ensure the task is properly cancelled even when the standard `task.cancel()` approach fails.
This approach has been suggested in this issue: https://github.com/pipecat-ai/pipecat/issues/1919#issuecomment-2921498074
And it seems to be working.